### PR TITLE
Add Notepad++ port from windows

### PR DESCRIPTION
**Pull request by matthewyang204**
Notepad++ ported with porting kit

### CODE REVIEW

File: office/npp/Portfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+name                npp
+version             8.6.4
+categories          office
+maintainers         gmail.com:matthew.yang204
+description         Notepad++ ported to macOS
+long_description    Notepad++ ported to macOS, with all features working. Supports all Windows shortcuts. Update system works.
+platforms           darwin
+license             ShareAlike
+homepage            https://github.com/matthewyang204/NPP-on-Mac
+master_sites        https://github.com/matthewyang204/NPP-on-Mac/releases/download/${version}/Notepad++.zip
```
Comment (aeiouaeiouaeiouaeiouaeiouaeiou) [inline office/npp/Portfile:11]: Is there any particular reason why the source code is not published? I'm just wondering what actually goes into a package that is over 400MB in size.

**Comment by reneeotten** [comment]
I doubt that we will want to add this to MacPorts. It is intended to compile software from source not to install pre-compiled binaries. End-user can just download the archive themselves from upstream' website.

File: office/npp/Portfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
+PortSystem          1.0
+name                npp
+version             8.6.4
+categories          office
+maintainers         gmail.com:matthew.yang204
+description         Notepad++ ported to macOS
+long_description    Notepad++ ported to macOS, with all features working. Supports all Windows shortcuts. Update system works.
+platforms           darwin
```
Comment (barracuda156) [inline office/npp/Portfile:8]: This is the default, so unneeded.

**Comment by barracuda156** [comment]
@matthewyang204 I do not think this app works on every macOS from 10.4 up and for all archs (including `ppc`, `ppc64` and `i386`). If not, `supported_archs` should be set, and platforms should reflect actually supported platforms.

**Comment by ShikiSuen** [comment]
@aeiouaeiouaeiouaeiouaeiouaeiou
This PR attempts to push a wine-wrapped WinNT app to a package manager dedicated for macOS.
Sigh.

P.S.: NotePad-- has native macOS version, providing similar experiences to NotePad++:
https://github.com/cxasm/notepad--
However, this is a Qt app, and its compilation process looks a little bit sophisticated.

**Comment by pmetzger** [comment]
@ShikiSuen MacPorts handles many things with "sophisticated" build systems.

**Comment by ryandesign** [comment]
> ported with porting kit

Thank you for contributing but Windows software wrapped in a Wine layer with Porting Kit is not the kind of thing we want in MacPorts.

**Comment by ShikiSuen** [comment]
I have just read this news indicating that some individuals are deliberately spreading virus-contaminated wine-bottled NotePad++ on the internet:
https://www.landiannews.com/archives/102878.html?utm_sources=ourl.co&utm_medium=social&utm_campaign=none

**Comment by kencu** [comment]
This is certainly a very very strong argument against allowing MacPorts to install any prebuilt binaries that don't come from our own build systems, built from open-source software, or suitably unassailably verified sources.

**Pull request closed without merging**